packages/telegram.rb

```diff
@@ -0,0 +1,24 @@
+require 'package'
+
+class Telegram < Package
+  description "Telegram is a messaging app with a focus on speed and security, it's super-fast, simple and free."
+  homepage 'https://telegram.org/'
+  version '2.5.1'
+  compatibility 'x86_64'
+  case ARCH
+  when 'x86_64'
+    source_url 'https://updates.tdesktop.com/tlinux/tsetup.2.5.1.tar.xz'
+    source_sha256 '939275a84498dedb1a33ac3f0242b0211548ddb087f9801518db92c5d6dc0151'
+    depends_on 'sommelier'
```

Comment:
Just take a try on it:) It works with `Qt` native Wayland backend (`QT_QPA_PLATFORM=wayland-egl DISPLAY= telegram`)

Reply:
Missing `qtbase` dependency?

Reply:
Sigh. We really need a way to do ```depends_on 'a' or 'b'```.

Reply:
Also agree on avoiding a sommelier dependency where the system wayland server will suffice... but does that work on i686? I imagine the wayland server there is pretty old? Or can we update that somehow?

Reply:
@supechicken @satmandu: I have removed the `sommelier` dependency.   Apparently, the `qtbase` dependency is not needed.